core/src/main/java/org/apache/iceberg/rest/Endpoint.java

```diff
@@ -46,6 +46,8 @@ public class Endpoint {
       Endpoint.create("POST", ResourcePaths.V1_NAMESPACE_PROPERTIES);
   public static final Endpoint V1_DELETE_NAMESPACE =
       Endpoint.create("DELETE", ResourcePaths.V1_NAMESPACE);
+  public static final Endpoint V1_NAMESPACE_EXISTS =
+      Endpoint.create("HEAD", ResourcePaths.V1_NAMESPACE);
```

Comment:
I've opened https://github.com/apache/iceberg/pull/11761 to actually do a HEAD request when checking for namespace existence

Reply:
@ajreid21 https://github.com/apache/iceberg/pull/11761 and https://github.com/apache/iceberg/pull/11760 landed now on main. Could you please rebase and add the `Endpoint.check` calls for view/namespace existence checks?

Reply:
I rebased and added the checks